### PR TITLE
docs(nx-dev): diataxis restructure

### PR DESCRIPTION
**Pull request by isaacplmann**
Big restructure of the docs.

Creates 5 main sections of the docs:

- Intro/tutorials
- Features (intro to each feature)
- Concepts (deeper explanations)
- Recipes (focused how-to guides)
- Reference

Later, after this PR:
- Integrated Repos and Project-Based Repos
- Update Create Your Own Plugin guide
- Automate showing concepts, recipes and reference properties by feature/category
- Deprecated section of the All Recipes list (will be visible in search)
  - workspace.json
  - tslint
  - workspace generators
  - workspace executors

### CODE REVIEW

File: docs/shared/core-features/automate-updating-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+The common case is that you will simply apply all migrations from the generated JSON file, exactly as they were generated in the previous step, by running:
+
+```bash
+nx migrate --run-migrations
+```
+
+This will update your source code in your workspace in accordance with the implementation of the various migrations which ran and all the changes will be unstaged ready for you to review and commit yourself.
+
+### Step 3: Cleaning up
```
Comment (juristr) [inline docs/shared/core-features/automate-updating-dependencies.md:56]: Do we actually recommend removing the `migrations.json` file? In many cases it is actually useful to have it (especially in larger workspaces), s.t. open PRs can rebase with the latest (already migrated main branch) and also re-run the migrations on the potentially not migrated source code.

File: docs/shared/core-features/enforce-project-boundaries.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+```
+
+## Tags
+
+Nx comes with a generic mechanism for expressing constraints: tags.
+
+First, use your project configuration (in `project.json` or `package.json`) to annotate your projects with `tags`. In this example, we will use three tags: `scope:client`. `scope:admin`, `scope:shared`.
+
+```jsonc
+// project "client"
```
Comment (juristr) [inline docs/shared/core-features/enforce-project-boundaries.md:41]: can we also have these as tabs? Showing a `project.json` excerpt for every project in a single tab? 
Having the code this way might be hard to understand for someone not knowing the project.json file structure well enough.

File: docs/shared/core-features/enforce-project-boundaries.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+  "tags": ["scope:shared"]
+}
+```
+
+Next you should update your root lint configuration:
+
+- If you are using **ESLint** you should look for an existing rule entry in your root `.eslintrc.json` called `"@nrwl/nx/enforce-module-boundaries"` and you should update the `"depConstraints"`:
+
+```jsonc
```
Comment (juristr) [inline docs/shared/core-features/enforce-project-boundaries.md:104]: in the code-examples I always prefer to have the file-name at the top as a comment, or maybe Ben's plugin allows to have that specified somehow differently, like Docusaurus allows you to do.

@bcabanes ?

File: docs/shared/plugin-features/use-code-generators.md [hunk 4]
```diff
@@ -1,18 +1,16 @@
-# Using Generators
+# Use Code Generators
 
 ## Overview
```
Comment (juristr) [inline docs/shared/plugin-features/use-code-generators.md:3]: I would just remove the title `## Overview` here and have the paragraph as the intro to the doc

File: docs/shared/core-features/enforce-project-boundaries.md (hunk 3 above)
Comment (bcabanes) [reply docs/shared/core-features/enforce-project-boundaries.md:107]: You can add it as a comment.

**Comment by isaacplmann** [comment]
Will merge this first thing Monday morning

**Pull request merged**